src/t38_gateway.c

```diff
@@ -1282,11 +1282,27 @@ static int process_rx_data(t38_core_state_t *t, void *user_data, int data_type,
            of the incrementing sequence numbers. We need to filter them here in a context sensitive manner. */
         if (t->current_rx_data_type != data_type  ||  t->current_rx_field_type != field_type)
         {
-            if (hdlc_buf->contents != (data_type | FLAG_DATA))
```

Comment:
you could keep this `if` and then do 
`if (t->current_rx_data_type != data_type) { span_log...}
else { span_other_log...` ?

Reply:
Good point